### PR TITLE
chore: move smoke test to foundry

### PR DESCRIPTION
**Pull request by nbaztec**
# What :computer: 
* Moves smoke test to `foundry-zksync`
* Supports smoke-test for:
  - Console logs
  - Normal test
  - `testFail` passes
  - Test failures
  - Fuzz testing

# Why :hand:
* Closer to the unit under test

# Evidence :camera:
Test passes in CI

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -93,30 +93,27 @@ jobs:
...
       
       - name: Install Rust
         uses: actions-rust-lang/setup-rust-toolchain@v1
         with:
           toolchain: nightly-2023-09-21
       
+      - uses: Swatinem/rust-cache@v2
+        with:
+          cache-on-failure: true
+
```
Comment (dutterbutter) [inline .github/workflows/test.yml:112]: I believe `actions-rust-lang/setup-rust-toolchain` actually configures this by default. https://github.com/actions-rust-lang/setup-rust-toolchain?tab=readme-ov-file#inputs

**Review by dutterbutter — APPROVED** [review]
Just the one comment re: cache.

File: .github/workflows/test.yml (hunk 1 above)
Comment (nbaztec) [reply .github/workflows/test.yml]: Crazy think it's not working as intended maybe since all runs take ~ 17minutes to finish

Comment (nbaztec) [reply .github/workflows/test.yml]: image

That seems too small of a size to realistically cache anything

**Pull request merged**